src/_xoConnection.js

```diff
@@ -0,0 +1,169 @@
+/* eslint-env jest */
+import defer from "golike-defer";
+import Xo from "xo-lib";
+import XoCollection from "xo-collection";
+import { find, forOwn } from "lodash";
+
+import config from "./_config";
+
+const logError = error => console.error(error);
+
+class XoConnection extends Xo {
+  constructor(opts) {
+    super(opts);
+
+    this._objects = new XoCollection();
+    this._watchers = {};
+    this._tempResourcesByClass = {
+      backupNg: [],
+      job: [],
+      user: [],
+    };
+  }
+
+  get objects() {
+    return this._objects;
+  }
+
+  _addObjectsListener() {
+    this.on("notification", ({ method, params }) => {
+      if (method !== "all") {
+        return;
+      }
+
+      const { _objects: objects, _watchers: watchers } = this;
+      const fn = params.type === "exit" ? objects.unset : objects.set;
+      forOwn(params.items, (item, id) => {
+        fn.call(objects, id, item);
+
+        const watcher = watchers[id];
+        if (watcher !== undefined) {
+          watcher(item);
+          delete watchers[id];
+        }
+      });
+    });
+  }
```

Comment:
Why not move in constructor?

Reply:
I create a method instead for readability  and for add the listener on connection not on instancing the class.

Reply:
IMHO it's simpler in the constructor.